server/src/modules/campaign-monitor/campaign-monitor.service.ts

```diff
@@ -593,15 +596,32 @@ export default class CampaignMonitorService {
 	 * @param listIds The IDs of all the campaign monitor lists we want to get all subscriber data from.
 	 */
 	static async getAllSubscribersFromLists(listIds: string[]): Promise<ICMActiveSubscriber[][]> {
-		// TODO: store each item in an individual file
-		return Promise.all(listIds.map((listId) => this.getAllSubscribersFromList(listId)));
+		return Promise.all(listIds.map((listId) => this.getAllSubscribersFromList(listId))).then(
+			(result) => {
+				listIds.forEach((listId, index) =>
+					this.createJSONFile(result[index], `campaign-monitor/${listId}`)
+				);
+				return result;
+			}
+		);
```

Comment:
I personally prefer the async await syntax over the .then() syntax.
It usually is shorter with less indentations and easier to read:
```suggestion
		const result = await Promise.all(listIds.map((listId) => this.getAllSubscribersFromList(listId)));
		listIds.forEach((listId, index) =>
			this.createJSONFile(result[index], `campaign-monitor/${listId}`)
		);
		return result;
```